### PR TITLE
ROC-3285 Integrate logging with Azure Application Insights

### PR DESCRIPTION
**Pull request by dharmendrak**
Jira: https://tools.hmcts.net/jira/browse/ROC-3285

Integrate nodejs-logging with Azure App Insights SDK

PRs effected by this change:
https://github.com/hmcts/expressjs-template/pull/21
https://github.com/hmcts/cmc-citizen-frontend/pull/517

### CODE REVIEW

File: log/Logger.js [hunk 1]
```diff
@@ -1,123 +1,24 @@
-const merge = require('lodash.merge');
```
[the remark this answers is not in the document]
Comment (dharmendrak) [reply log/Logger.js]: yes, there are, however didn't push PR so far as it need nodes-logging  latest version

File: README.md [hunk 2]
```diff
@@ -3,13 +3,17 @@
 [![Greenkeeper badge](https://badges.greenkeeper.io/hmcts/nodejs-logging.svg)](https://greenkeeper.io/)
 
 A logging component used by Reform's Node.js applications. Some background info:
```
Comment (geek-soft) [inline README.md]: I would recommend adding header note (in bold) saying that this is not compatible with reform tactical logging spec and that for tactical applications logger 2.x should be used.

File: README.md [hunk 3]
```diff
@@ -25,25 +29,31 @@ Require it:
...
+```
+logger.log({
+  level: 'info',
+  message: 'What time is the testing at?'
+});
+
+ Outputs:
+{ level: 'info',
+  message: 'What time is the testing at?',
+  label: 'right meow!',
```
Comment (geek-soft) [inline README.md]: Where did label come from? Shouldn't that be `app.js` as you instantiated above?

File: log/Logger.js [hunk 4]
```diff
@@ -1,123 +1,24 @@
...
-  return !str || str.trim().length === 0;
+const moment = require('moment')
+const winston = require('winston')
+
+const container = new winston.Container()
+const tsFormat = () => (moment().format('YYYY-MM-DDTHH:mm:ssZ'))
+
+function transports (label) {
+  return [new winston.transports.Console({
+    level: process.env.LOG_LEVEL || 'INFO',
```
Comment (geek-soft) [inline log/Logger.js]: Log level has to be lower case. Default 'INFO' causes that logs are not being produced at all.

I will fix that

**Comment by geek-soft** [comment]
@timja-kainos, @rlewan I removed request tracing and polished it a bit. Would be helpful to do quick final review.

**Pull request merged**